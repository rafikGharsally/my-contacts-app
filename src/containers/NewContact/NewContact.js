import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { AddContact, ContactForm } from '../../components/GenericPages/Contact';
import { addNewContact } from '../../actions/contact';

class NewContact extends Component {

  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
  }

  submit(contact) {

    const { addNewContact, history } = this.props;
    addNewContact(contact).then(success => {
     console.log('!success added a new contact---', success);
     history.push('/');
    }, error => {
      console.log('error adding a new contact---', error);
    })
  }

  render() {
    return (
      <AddContact title="Add contact" previousState="/">
        <ContactForm submitForm={this.submit} />
      </AddContact>

    );
  }
}

const mapStateToProps = state => {
  return {
  }
};


const mapDispatchToProps = dispatch => {
  return bindActionCreators({ addNewContact }, dispatch);
};

export default connect(
  mapStateToProps, mapDispatchToProps)(NewContact);
