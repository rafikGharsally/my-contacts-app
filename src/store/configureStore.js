import { createStore, applyMiddleware, compose } from 'redux';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
import axios from 'axios';
import axiosMiddleware from 'redux-axios-middleware';
import thunk from 'redux-thunk';
import rootReducer from '../reducers'

export default function createReduxStore() {
  // add a browser history
  const history = createHistory();

  // intercept and dispatch action here
  const middleware = routerMiddleware(history);

  const client = axios.create();

  const clientOptions = {
    returnRejectedPromiseOnError: true,
    errorSuffix: '_ERROR',
    successSuffix: '_SUCCESS'
  };

  // Add the reducer to your store on the `router` key
  // Also apply our middleware for navigating
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const store = createStore(
    rootReducer,
    // initialState
    {},
    composeEnhancers(applyMiddleware(middleware, thunk, axiosMiddleware(client, clientOptions))
    ));

  if (process.env.NODE_ENV !== "production") {
    if (module.hot) {
      module.hot.accept('../reducers', () => {
        const newRootReducer = require('../reducers').default;
        store.replaceReducer(newRootReducer);
      });
    }
  }

  return store;
};