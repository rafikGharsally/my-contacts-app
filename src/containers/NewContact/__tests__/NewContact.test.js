import React from 'react';
import NewContact  from '../NewContact';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';


// Create the mock store
import configureMockStore from 'redux-mock-store';
const mockStore = configureMockStore();

const store = configureStore([
  thunk,
])();

describe('NewContact', () => {
  let wrapper, store;

  beforeEach(() => {
    const initialState = {
      contact: {
        data: {"id":1,"name":"Rafik dragon"},
        isFetching:true,
        error:false
      }
    };
    store = mockStore(initialState);
    // Shallow render the container passing in the mock store
    wrapper = shallow(
      <NewContact store={store} />
    );
  });

  it('should be truthy', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('renders and matches our snapshot', () => {
    const component = renderer.create((<Router><NewContact store={store} /></Router>));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});



