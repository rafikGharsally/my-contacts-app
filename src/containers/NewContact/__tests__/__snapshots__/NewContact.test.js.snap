// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewContact renders and matches our snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root-12 MuiPaper-elevation4-18 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-10"
  >
    <div
      className="MuiToolbar-root-39 MuiToolbar-gutters-40"
    >
      <button
        className="MuiButtonBase-root-47 MuiIconButton-root-41"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-46"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-50 AddContact-icon-2"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-56"
        />
      </button>
      <p
        className="MuiTypography-root-63 MuiTypography-body1-72 MuiTypography-colorInherit-82"
        style={
          Object {
            "flex": "1",
          }
        }
        type="title"
      >
        Add contact
      </p>
    </div>
  </header>
  <div
    className="sc-bwzfXH eJEMPj"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-98 MuiInputLabel-root-93 MuiInputLabel-formControl-94 MuiInputLabel-animated-97"
          data-shrink={false}
          htmlFor="name"
        >
          Name
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104 MuiInput-underline-107"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-111"
            defaultValue={undefined}
            disabled={false}
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-98 MuiInputLabel-root-93 MuiInputLabel-formControl-94 MuiInputLabel-animated-97"
          data-shrink={false}
          htmlFor="job_title"
        >
          job_title
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104 MuiInput-underline-107"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-111"
            defaultValue={undefined}
            disabled={false}
            id="job_title"
            name="job_title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-98 MuiInputLabel-root-93 MuiInputLabel-formControl-94 MuiInputLabel-animated-97"
          data-shrink={false}
          htmlFor="address"
        >
          address
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104 MuiInput-underline-107"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-111"
            defaultValue={undefined}
            disabled={false}
            id="address"
            name="address"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-98 MuiInputLabel-root-93 MuiInputLabel-formControl-94 MuiInputLabel-animated-97"
          data-shrink={false}
          htmlFor="phone"
        >
          phone
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104 MuiInput-underline-107"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-111"
            defaultValue={undefined}
            disabled={false}
            id="phone"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-98 MuiInputLabel-root-93 MuiInputLabel-formControl-94 MuiInputLabel-animated-97"
          data-shrink={false}
          htmlFor="email"
        >
          email
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-103 MuiInput-fullWidth-110 MuiInput-formControl-104 MuiInput-underline-107"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-111"
            defaultValue={undefined}
            disabled={false}
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="sc-bdVaJa ccfodj"
      >
        <button
          className="MuiButtonBase-root-47 MuiButton-root-116 MuiButton-contained-125 MuiButton-containedPrimary-126 MuiButton-raised-128 MuiButton-raisedPrimary-129 ContactForm-button-87"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="submit"
        >
          <span
            className="MuiButton-label-117"
          >
            Save
          </span>
          <span
            className="MuiTouchRipple-root-56"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
